Fix reported issues w.r.t python2.7 and some apple silicone quirks

jtrmal opened the pull request:
@danpovey  for your reviewing pleasure ;)

csukuangfj commented on src/util/parse-options-test.cc:
```diff
@@ -120,7 +120,7 @@ void UnitTestParseOptions() {
     po4.Read(argc4, argv4);
     assert(false); // Should not reach this part of code.
-  } catch(std::exception e) {
+  } catch(std::exception &e) {
```
I suggest using `const std::exception &e`.

`const` makes the intention clear that it is not intended to be modified.

jtrmal commented:
@csukuangfj  hi, good point, changed. I also fixed the failing tests while I was at it -- authors of the tests forgot that `KALDI_ASSERT(cond)` calls/evaluates `cond` twice. LMK

danpovey commented:
I'm surprised some of these issues lasted as long as they have, about the templates.  anyway looks OK to me.

jtrmal commented:
I'm surprised too, but some stuff we ignored and some stuff the older compilers didn't complain about... 🤷

boeddeker commented:
Is there a reason, that `mkdir -p $PWD/python` was removed in `check_dependencies`?
My kaldi install script is currently failing, and I get `ln: failed to create symbolic link '.../kaldi/tools/python/python'*` warnings. I will fix it in my script, but I assume it would be good to have check_dependencies to be robust.

jtrmal commented:
Might had been unintentional change, are you willing to create a PR?
Thanks!
y.

On Thu, Aug 28, 2025 at 11:29 PM Christoph Boeddeker <
***@***.***> wrote:

> *boeddeker* left a comment (kaldi-asr/kaldi#4924)
> <https://github.com/kaldi-asr/kaldi/pull/4924#issuecomment-3234995818>
>
> Is there a reason, that mkdir -p $PWD/python was removed in
> check_dependencies?
> My kaldi install script is currently failing, and I get ln: failed to
> create symbolic link '.../kaldi/tools/python/python'* warnings. I will
> fix it in my script, but I assume it would be good to have
> check_dependencies to be robust.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/4924#issuecomment-3234995818>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUKYX3RRBMGB4RZ5NJXEQ33P5YDZAVCNFSM6AAAAACFCWEGP2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTEMZUHE4TKOBRHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
